src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -854,42 +915,25 @@ class Simplex_tree {
   */
   template <class RandomVertexHandleRange = std::initializer_list<Vertex_handle>>
   std::pair<Simplex_handle, bool> insert_simplex_raw(const RandomVertexHandleRange& simplex,
-                                                     Filtration_value filtration) {
+                                                     const Filtration_value& filtration) {
     Siblings * curr_sib = &root_;
     std::pair<Simplex_handle, bool> res_insert;
     auto vi = simplex.begin();
+
     for (; vi != std::prev(simplex.end()); ++vi) {
       GUDHI_CHECK(*vi != null_vertex(), "cannot use the dummy null_vertex() as a real vertex");
-      res_insert = curr_sib->members_.emplace(*vi, Node(curr_sib, filtration));
-      if (res_insert.second) {
-        // Only required when insertion is successful
-        update_simplex_tree_after_node_insertion(res_insert.first);
-      }
-      if (!(has_children(res_insert.first))) {
-        res_insert.first->second.assign_children(new Siblings(curr_sib, *vi));
-      }
+      res_insert = insert_node_<false, true, false>(curr_sib, *vi, filtration);
```

Comment:
I agree... But I found it better than having 4 methods doing more or less the same with just small variations. The other solution would have been to just keep the `try_emplace` and the `update_simplex_tree_after_node_insertion` in the method and letting the rest in the calling methods, but this would have lessen the point of `insert_node_`. Methods like `insert_simplex_raw` are so much more readable now.
If someone has a nicer solution, they are welcome!